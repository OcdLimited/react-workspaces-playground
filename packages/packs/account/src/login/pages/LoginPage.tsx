import React, { useState } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useTranslation } from 'react-i18next';
import { TenancyBox, LoginForm } from '../components';

interface LoginPageProps {
	onSubmit: any;
	isMultiTenant: boolean;
	isSelfRegistrationEnabled: boolean;
	autoFocus: boolean;
	onTenantChanged?: any;
}

export function LoginPage({ onSubmit, isMultiTenant, autoFocus, onTenantChanged }: LoginPageProps) {
	const { t } = useTranslation('AbpAccount');

	const loginSchema = Yup.object().shape({
		username: Yup.string().required(t('The {0} field is required.').replace('{0}', t('DisplayName:Username'))),
		password: Yup.string().required(t('The {0} field is required.').replace('{0}', t('DisplayName:Password'))),
	});

	return (
		<React.Fragment>
			<TenancyBox visible={isMultiTenant} tenantName="" onTenantChanged={onTenantChanged} />
			<Formik
				initialValues={{
					username: '',
					password: '',
				}}
				validationSchema={loginSchema}
				onSubmit={onSubmit}
			>
				{props => <LoginForm autoFocus={autoFocus} {...props} />}
			</Formik>
		</React.Fragment>
	);
}

LoginPage.defaultProps = {
	autoFocus: false,
};

export default LoginPage;
