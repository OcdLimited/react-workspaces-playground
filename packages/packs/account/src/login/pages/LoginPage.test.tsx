import React from 'react';
import { render } from '@testing-library/react';
import { LoginPage } from './LoginPage';

it('snapshot', () => {
	const { container } = render(<LoginPage />);
	expect(container).toMatchSnapshot();
});
