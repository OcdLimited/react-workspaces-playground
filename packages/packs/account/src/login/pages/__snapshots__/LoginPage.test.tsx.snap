// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5"
    >
      Login
    </h1>
    <form
      action="#"
      class="makeStyles-form-5"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          UserNameOrEmailAddress
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="username"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="username"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-141"
            >
              <span>
                UserNameOrEmailAddress
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
          for="password"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-140"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-174 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-177"
              data-indeterminate="false"
              name="remember"
              type="checkbox"
              value="false"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          RememberMe
        </span>
      </label>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
            href="#"
          >
            AreYouANewUser
             
            Register
          </a>
        </div>
      </div>
    </form>
  </div>
</div>
`;
